src/Operation/Wrap.php

```diff
@@ -16,25 +16,20 @@
 final class Wrap extends AbstractOperation
 {
     /**
-     * @return Closure(iterable<TKey, T>): Generator<int, array<array-key, T>>
+     * @return Closure(iterable<TKey, T>): Generator<int, non-empty-array<TKey, T>>
      */
     public function __invoke(): Closure
     {
-        /** @var Closure(iterable<TKey, T>): Generator<int, array<array-key, T>> $pipe */
-        $pipe = (new Pipe())()(
-            (new Map())()(
-                /**
-                 * @param T $value
-                 * @param array-key $key
-                 *
-                 * @return array<array-key, T>
-                 */
-                static fn (mixed $value, int|string $key): array => [$key => $value]
-            ),
-            (new Normalize())()
-        );
-
-        // Point free style.
-        return $pipe;
+        return
+        /**
+         * @param iterable<TKey, T> $iterable
+         *
+         * @return Generator<int, non-empty-array<TKey, T>>
```

Comment:
this will not always be a `non-empty-array` I think? What if the `iterable` is empty?